### PR TITLE
Feature/call act/scoping

### PR DESCRIPTION
**Pull request by rallen2010**
- Adds scoped Variables to engine
- Adds InVariables and outVariables to callActivities
- Adds basic tests (utplsql) for callActivities
- fixes issue #451 to allow trailing ';' on SQL-typed variable. expressions

### CODE REVIEW

File: src/plsql/flow_call_activities.pkb [hunk 1]
```diff
@@ -122,13 +114,6 @@ as
        for update of prdg.prdg_diagram_level wait 2
     ;
 
-    /*l_called_dgrm_id := flow_diagram.get_current_diagram
```
Comment (ralfmueller) [inline src/plsql/flow_call_activities.pkb:125]: intentionally commented out?

File: src/common_db.sql [hunk 2]
```diff
@@ -81,6 +83,7 @@ PROMPT >> Page Views
 @views/engine-app/flow_p0002_diagrams_vw.sql
 @views/engine-app/flow_p0007_vw.sql
 @views/engine-app/flow_p0007_instances_counter_vw.sql
+@views/engine-app/flow_p0007_vw.sql
```
Comment (commi235) [inline src/common_db.sql:86]: double call?

File: src/deinstall_objects.sql [hunk 3]
```diff
@@ -48,6 +49,7 @@ PROMPT >> Views
 drop view flow_p0002_diagrams_vw;
 drop view flow_p0007_vw;
 drop view flow_p0007_instances_counter_vw;
+drop view flow_p0007_vw;
```
Comment (commi235) [inline src/deinstall_objects.sql:52]: duplicate?

**Review by ralfmueller — APPROVED** [review]
Not checked all the details here as I'm not that familiar with the implementation details of Flows but got the intention with introducing scope and how that impacts process variables. Maybe Moritz can have a more detailed look into this

File: src/plsql/flow_call_activities.pkb (hunk 1 above)
Comment (rallen2010) [reply src/plsql/flow_call_activities.pkb:125]: Yes.  Earlier on in the development we switched from evaluating the called diagram when the callActivity was current object TO evaluating which diagrams would get called for the whole process at process creation.  This populates the flow_instance_diagrams table at process creation.   So no need to evaluate mid-process execution.  The changes on this branch merely took the old code, which was commented out, and deleted it.

File: src/common_db.sql (hunk 2 above)
Comment (rallen2010) [reply src/common_db.sql:86]: Looks like it!

File: src/deinstall_objects.sql (hunk 3 above)
Comment (rallen2010) [reply src/deinstall_objects.sql]: yes - resolved

File: src/common_db.sql (hunk 2 above)
Comment (rallen2010) [reply src/common_db.sql]: fixed

**Review by commi235 — APPROVED** [review]

**Pull request merged**